client/state/notices/middleware.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+import { truncate } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { successNotice, errorNotice } from 'state/notices/actions';
+import { getSitePost } from 'state/posts/selectors';
+import {
+	POST_DELETE_FAILURE,
+	POST_DELETE_SUCCESS,
+	POST_SAVE_SUCCESS
+} from 'state/action-types';
+
+export const OBSERVERS = {
+	[ POST_DELETE_FAILURE ]: ( action, dispatch, getState ) => {
+		const post = getSitePost( getState(), action.siteId, action.postId );
+
+		let message;
+		if ( post ) {
+			message = translate( 'An error occurred while deleting "%s"', {
+				args: [ truncate( post.title, { length: 24 } ) ]
+			} );
+		} else {
+			message = translate( 'An error occurred while deleting the post' );
+		}
+
+		dispatch( errorNotice( message ) );
+	},
+	[ POST_DELETE_SUCCESS ]: ( action, dispatch ) => {
+		dispatch( successNotice( translate( 'Post successfully deleted' ) ) );
+	},
+	[ POST_SAVE_SUCCESS ]: ( action, dispatch ) => {
+		let text;
+		switch ( action.post.status ) {
+			case 'trash':
+				text = translate( 'Post successfully moved to trash' );
+				break;
+
+			case 'publish':
+				text = translate( 'Post successfully published' );
+				break;
+		}
+
+		if ( text ) {
+			dispatch( successNotice( text ) );
+		}
+	}
```

Comment:
>Isn't this one of the very reasons we use functions [...] Maybe it's more about the extent to which we do this.

Yes to both of these. It's a balance. I don't find your suggestion here to be particularly extreme, but I do think some of the [earlier discussion](https://github.com/Automattic/wp-calypso/pull/6645#discussion_r70321190) was starting to head in the direction of jumping the gun on abstractions (decorators, `createActionHandler`). My thought with keeping it as a single object was to keep it simple to start, since I'm honestly unsure how this will evolve (https://github.com/Automattic/wp-calypso/pull/6645#issuecomment-232050396). This may or may not scale very well, but there's not been an obvious alternative presented that would be more readable/maintainable.

>With the object wrapping around it, we're testing both the functionality and the calling convention.

Isn't this true of any variable? A variable's name is its calling convention, so I still don't see why we're any more tied to `HANDLERS[ POST_SAVE_SUCCESS ]` than we'd be to `handlePostSaveSuccess`.